### PR TITLE
 Update normal equation developments with the new CSR API 

### PR DESCRIPTION
**Pull request by nychiang**
The base of this PR is #519, NOT `develop`.
This PR use the latest CSR API in normal equation development.

### CODE REVIEW

**Comment by cnpetra** [comment]
I missed that the base for this PR is `vector_delta_dev` and I merged `vector_delta_dev` branch. Can this be rebased on develop?

File: src/Optimization/hiopKKTLinSysSparseNormalEqn.cpp [hunk 1]
```diff
@@ -144,28 +171,95 @@ bool hiopKKTLinSysSparseNormalEqn::build_kkt_matrix(const hiopVector& delta_wx_i
   assert(nineq == Dd_->get_size());
   assert(nx == Dx_->get_size());
 
+  // NOTE:
+  // hybrid compute mode -> linear algebra objects used internally by the class will be allocated on the device. Most of the inputs
+  // to this class will be however on HOST under hybrid mode, so some objects are copied/replicated/transfered to device
+  // gpu copute mode -> not yet supported
+  // cpu compute mode -> all objects on HOST, however, some objects will still be copied (e.g., Hd_) to ensure code homogeinity
```
Comment (cnpetra) [inline src/Optimization/hiopKKTLinSysSparseNormalEqn.cpp:178]: typo "homogeinity" -> homogeneity

**Review by cnpetra — APPROVED** [review]
lgtm, just two minor issues

**Comment by nychiang** [comment]
@cnpetra rebased to develop and addressed your comments

**Pull request merged**